bin/interactions/fun/xkcd.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:async';
+
+import 'package:dio/dio.dart';
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_interactions/interactions.dart';
+
+import '../../obsidian_dart.dart';
+import '../../utils/constants.dart';
+import '../../utils/embed.dart';
+
+class FunXkcdInteractions {
+  final BASE_URL = 'https://xkcd.com';
+
+  FunXkcdInteractions() {
+    botInteractions.registerSlashCommand(
+      SlashCommandBuilder(
+        'xkcd',
+        'XKCD group of commands.',
+        [
+          CommandOptionBuilder(CommandOptionType.subCommand, 'latest',
+              'Retrives lastest XKCD comic.')
+            ..registerHandler(xkcdLatestSlashCommand),
+          CommandOptionBuilder(
+            CommandOptionType.subCommand,
+            'search',
+            'Retrives XKCD comic.',
+            options: [
+              CommandOptionBuilder(
+                  CommandOptionType.integer, 'comic', 'XKCD Comic number.',
+                  required: true)
+            ],
+          )..registerHandler(xkcdComicSlashCommand),
+        ],
+      ),
+    );
+  }
+
+  EmbedBuilder xkcdEmbed(SlashCommandInteractionEvent event, String title,
+      String desc, String imgUrl, String footerText) {
+    return EmbedBuilder()
+      ..title = title
+      ..description = desc
+      ..color = DiscordColor.sapGreen
+      ..imageUrl = imgUrl
+      ..addFooter((footer) {
+        footer.text = footerText;
+        footer.iconUrl = event.interaction.userAuthor?.avatarURL();
+      });
+  }
```

Comment:
- A `DateTime.now()` timestamp is better imo as all the other embeds use it.
- The embed footer can also be changed to match embeds in other commands.
- The XKCD comic number can be appended to the title and the date of publishing can be added to the embed description.